dms/security/ir.model.access.csv

```diff
@@ -3,18 +3,22 @@ id,name,model_id/id,group_id/id,perm_read,perm_write,perm_create,perm_unlink
 access_dms_tag_user,dms_tag_user,model_dms_tag,group_dms_user,1,1,1,1
 access_dms_category_user,dms_category_user,model_dms_category,group_dms_user,1,1,1,1
 
+access_dms_storage_public,dms_storage_public,model_dms_storage,base.group_public,1,0,0,0
```

Comment:
Why giving this permission (and others) to public/portal? I don't see that as a good default. You won't see this technique in any other hidden model.

Reply:
It's apply according to work fine some things:
- Prevent apply .`sudo() `in controller recordset.
- Prevent inconsistent data (if apply always `.sudo()` total elements in directories it's not correct because show total elements and not total elements according to request user can access).
- Prevent incorrect `check_access` when directories and files have storage with inherit permissions.

Reply:
But storages don't have permissions. The permissions are on directories. Storages should be still be accessed with `sudo`. If not, you are exposing a security hole. Anyone will be able to query via XML-RPC your storages definition. You can use `related` fields with `related_sudo=True` for avoiding the need of put `sudo` everywhere, and also `auto_join` when needed.

Reply:
I have checked to remove it (related to dms.storage) but maybe isn't possible because ```_get_directory_ids_with_res_model_without_access``` need to filter by (`"storage_id.inherit_access_from_parent_record", "=", True)`, and portal (for example) can't access or public user if we share directory link.
Any idea to solve it?

Reply:
As said, a related field with `related_sudo` and `auto_join`.